StardewArchipelago/Items/Unlocks/MagicUnlockManager.cs

```diff
@@ -0,0 +1,284 @@
+                ReceiveSpell("elemental:teleport");
+            }
+            return new LetterInformationAttachment(receivedItem);
+        }
+
+        private LetterAttachment SendShockwaveSpell(ReceivedItem receivedItem)
+        {
+            foreach (var farmer in Game1.getAllFarmers())
+            {
+                ReceiveSpell("nature:shockwave");
+            }
+            return new LetterInformationAttachment(receivedItem);
+        }
+
+        private LetterAttachment SendTendrilsSpell(ReceivedItem receivedItem)
+        {
+            foreach (var farmer in Game1.getAllFarmers())
+            {
+                ReceiveSpell("nature:tendrils");
+            }
+            return new LetterInformationAttachment(receivedItem);
+        }
+
+        private LetterAttachment SendLanternSpell(ReceivedItem receivedItem)
+        {
+            foreach (var farmer in Game1.getAllFarmers())
+            {
+                ReceiveSpell("nature:lantern");
+            }
+            return new LetterInformationAttachment(receivedItem);
+        }
+
+        private LetterAttachment SendPhotosynthesisSpell(ReceivedItem receivedItem)
+        {
+            foreach (var farmer in Game1.getAllFarmers())
+            {
+                ReceiveSpell("nature:photosynthesis");
+            }
+            return new LetterInformationAttachment(receivedItem);
+        }
+
+        private LetterAttachment SendMeteorSpell(ReceivedItem receivedItem)
+        {
+            foreach (var farmer in Game1.getAllFarmers())
+            {
+                ReceiveSpell("eldritch:meteor");
+            }
+            return new LetterInformationAttachment(receivedItem);
+        }
+
+        private LetterAttachment SendBloodmanaSpell(ReceivedItem receivedItem)
+        {
+            foreach (var farmer in Game1.getAllFarmers())
+            {
+                ReceiveSpell("eldritch:bloodmana");
+            }
+            return new LetterInformationAttachment(receivedItem);
+        }
+
+        private LetterAttachment SendLuckstealSpell(ReceivedItem receivedItem)
+        {
+            foreach (var farmer in Game1.getAllFarmers())
+            {
+                ReceiveSpell("eldritch:lucksteal");
+            }
+            return new LetterInformationAttachment(receivedItem);
+        }
+
+        private LetterAttachment SendSpiritSpell(ReceivedItem receivedItem)
+        {
+            foreach (var farmer in Game1.getAllFarmers())
+            {
+                ReceiveSpell("eldritch:spirit");
+            }
+            return new LetterInformationAttachment(receivedItem);
+        }
+
+        private LetterAttachment SendRewindSpell(ReceivedItem receivedItem)
+        {
+            foreach (var farmer in Game1.getAllFarmers())
+            {
+                ReceiveSpell("arcane:rewind");
+            }
+            return new LetterInformationAttachment(receivedItem);
+        }
+
+        private void ReceiveSpell(string spellName)
```

Comment:
couple of var available

Reply:
Not sure what you mean with this one.  if I replace object[] with var the code complains.  I think those are important.